Follow-up API design for #28389 (#29200)
fleetdm/fleet: docs/REST API/rest-api.md

}
```

### Get batch script summary

Get statuses and host counts for a batch-run script.

> `GET /api/v1/fleet/scripts/batch/summary`

Reviewer: Per [agreement with @sgress454](https://github.com/fleetdm/fleet/pull/29200/files#r2093811935) that this should be a path parameter
```suggestion
`GET /api/v1/fleet/scripts/batch/summary/:batch_execution_id`
```

> `GET /api/v1/fleet/scripts/batch/summary/:batch_execution_id`

Reviewer: And cherry pick to the RC before it goes out? Seems feasible and a bit cleaner yea.

There's a bit of a semantics issue though - the existing `POST` `.../scripts/batch` is for uploading scripts, unrelated to the batch-run functionality, and removing `summary` would make it slightly less clear that this is referring to a batch _execution_ and not a batch script upload, but it's already not _that_ clear, so...why not

> `GET /api/v1/fleet/scripts/batch/summary/:batch_execution_id`

Author: Makes sense to leave it then I think 👍

#### Parameters

| Name            | Type    | In   | Description                                                                                    |
| ----            | ------- | ---- | --------------------------------------------                                                   |

> | batch_execution_id | integer | body | **Required**. The ID returned from a batch script run. |

Reviewer: ```suggestion
| batch_execution_id | string | path | **Required**. The ID returned from a batch script run. |
```

#### Example

> `GET /api/v1/fleet/scripts/batch/summary`

Reviewer: ```suggestion
`GET /api/v1/fleet/scripts/batch/summary/abc-def`
```

> 

Reviewer: ```suggestion

```

##### Request body

```json
{

>   "batch_execution_id": "e797d6c6-3aae-11ee-be56-0242ac120002"

Reviewer: Seems like this should be as a query param since it's a GET?

Reviewer: or in the path?

}
```

##### Default response

`Status: 200`

```json
{
  "ran": 12345,
  "pending": 234,

>   "error": 18

Reviewer 2: @rachaelshaw couple things here:

1. I was able to get `canceled` in here; we don't have to necessarily display it in the UI at this stage but we can still document it in here
2. I think we do want to change `error` to `errored` because we have a standard `error` key in most of our API response structs representing "something went wrong with this API call"

Author: Makes sense, I'm cool with `errored` 👍

Reviewer 2: K I'll update here, + there's a couple other fields in there that I think help w/ visualization:
```suggestion
  "errored": 18,
  "canceled": 2,
  "targeted": 12599,
  "script_id": 555,
  "script_name": "my-script.sh",
  "team_id": 123
```

Reviewer: Confirming that `team_id` will come from the new summary endpoint and not be from the activity? Works just as well, want to make sure we're synced.

Reviewer: > errored

To be clear we are not changing the UI to say "Errored" right

Reviewer 2: > Confirming that `team_id` will come from the new summary endpoint and not be from the activity? Works just as well, want to make sure we're synced.

It looks like my PR put it in both -- I may have misread where you were asking for it.  I can back the activity one out if you don't think it's needed.

> To be clear we are not changing the UI to say "Errored" right

Correct -- no change in the UI, just in the API response

Reviewer: Either place would work, and I implemented the UI to grab it from this new endpoint so all good if you want to pull it out of the activity response
